### DISCUSSION TITLE
How to use envConfigMaps from Dagster K8s

### DISCUSSION THREAD

**Discussion by andrescolodrero** [dagster-io/dagster#19135] [Q&A]
Hi,
i have a job that use and env value, by example:
datagen = DataGeneratorResource(
    num_days=EnvVar.int("HACKERNEWS_NUM_DAYS_WINDOW"),
)

I tried to setup in the chart the env value, and the Run Pod get the correct value, but the job cant find the environmental value. I suposed, i need a ".env" file?
 
I also tried to setup the job with a key/value config map:
envConfigMaps: 
         - name: job-environment

But stil failing. do i have to mount the .env file?

**Comment by jmsanders** [comment]
There are a couple of different ways you can set environment variables in K8s:

https://docs.dagster.io/deployment/guides/kubernetes/deploying-with-helm#step-6-configure-your-user-deployment

The config you've posted assumes that there's a [K8s ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) already created named `job-environment`. You shouldn't need to mount any additional files, but if you choose to, you can mount a [K8s Volume](https://kubernetes.io/docs/concepts/storage/volumes/).
